libs/beacon/src/beacon_manager.cpp

```diff
@@ -600,7 +588,8 @@ bool BeaconManager::Verify(byte_array::ConstByteArray const &group_public_key,
     return false;
   }
 
-  return crypto::mcl::VerifySign(tmp, message, tmp2, group_g_);
+  Generator generator_tmp{generator_string};
```

Comment:
this needs to be tmp because of the static issue? Is there a high overhead to constructing a generator for every verify?

Reply:
Yes, the tmp is here as this function is static and the generator is no longer static. From some quick benchmarks it adds around 10% onto the verification time, which is approx 0.15ms.

Reply:
I am happy to stick with the solution in the deployment updates PR. This was something implemented before I saw that.